lib/features/support/interactive_cli.rb

```diff
@@ -0,0 +1,117 @@
+require 'open3'
+
+# Encapsulates a terminal session, retaining state and input streams for interactive tests
+class InteractiveCLI
+  # @!attribute [r] parsed_output
+  #   @return [Array] An array of output strings received from the terminals STDOUT pipe
+  attr_reader :parsed_output
```

Comment:
I'd be tempted to call this `stdout_lines` for clarity, as there's no really any parsing going on